helper/tests/provisioners/provisioners_acc_test.go

```diff
@@ -0,0 +1,142 @@
+			continue
+		}
+		builders = append(builders, k)
+	}
+
+	var provisioners []string
+	for k := range ProvisionersAccTest {
+		if p != "all" && !strings.Contains(p, k) {
+			continue
+		}
+		provisioners = append(provisioners, k)
+	}
+
+	// build template file and run build
+	for _, builder := range builders {
+		builderAcc := BuildersAccTest[builder]
+		builderConfig, err := builderAcc.GetConfig()
+		if err != nil {
+			t.Fatalf("bad: failed to read builder config: %s", err.Error())
+		}
+		// Run a build for each builder with each of the provided provisioners
+		for _, provisioner := range provisioners {
+			testName := fmt.Sprintf("testing %s builder against %s provisioner", builder, provisioner)
+			t.Run(testName, func(t *testing.T) {
+				provisionerAcc := ProvisionersAccTest[provisioner]
+				provisionerConfig, err := provisionerAcc.GetConfig()
+				if err != nil {
+					t.Fatalf("bad: failed to read provisioner config: %s", err.Error())
+				}
+
+				// Write json template
+				out := bytes.NewBuffer(nil)
+				fmt.Fprintf(out, `{"builders": [%s],"provisioners": [%s]}`, builderConfig, provisionerConfig)
+				fileName := fmt.Sprintf("%s_%s.json", builder, provisioner)
+				filePath := filepath.Join("./", fileName)
+				writeJsonTemplate(out, filePath, t)
+
+				// set pre-config with necessary builder and provisioner
+				c := buildCommand(t, builderAcc, provisionerAcc)
+				args := []string{
+					filePath,
+				}
+
+				err = provisionerAcc.RunTest(c, args)
+				if err != nil {
+					// Cleanup created resources
+					testshelper.CleanupFiles(fileName)
+					builderAcc.CleanUp()
+					t.Fatalf("bad: failed to to run build: %s", err.Error())
+				}
+
+				// Cleanup created resources
+				testshelper.CleanupFiles(fileName)
+				err = builderAcc.CleanUp()
+				if err != nil {
+					t.Fatalf("bad: failed to clean up resources: %s", err.Error())
+				}
+			})
+		}
+	}
+}
+
+func provisionerAccTestPreCheck(t *testing.T) (string, string) {
+	b := os.Getenv("ACC_TEST_BUILDERS")
+	p := os.Getenv("ACC_TEST_PROVISIONERS")
+	// validate if we want to run provisioners acc tests
+	if b == "" || p == "" {
+		t.Skip("Provisioners Acceptance tests skipped unless env 'ACC_TEST_BUILDERS' and 'ACC_TEST_PROVISIONERS' are set")
+	}
+	return b, p
+}
+
+func writeJsonTemplate(out *bytes.Buffer, filePath string, t *testing.T) {
+	outputFile, err := os.Create(filePath)
+	if err != nil {
+		t.Fatalf("bad: failed to create template file: %s", err.Error())
+	}
+	_, err = outputFile.Write(out.Bytes())
+	if err != nil {
+		t.Fatalf("bad: failed to write template file: %s", err.Error())
+	}
+	outputFile.Sync()
+}
+
+func buildCommand(t *testing.T, builder BuilderAcceptance, provisioner ProvisionerAcceptance) *command.BuildCommand {
+	c := &command.BuildCommand{
+		Meta: testshelper.TestMetaFile(t),
+	}
+	c.CoreConfig.Components.BuilderStore = builder.GetBuilderStore()
+	c.CoreConfig.Components.ProvisionerStore = provisioner.GetProvisionerStore()
```

Comment:
@nywilken Moved these two configs to the builder and provisioned to make it more generic. Was it something you were thinking about?

Reply:
This is nice. I like this refactor. I created a draft PR to illustrate what I was thinking https://github.com/hashicorp/packer/pull/8985

But maybe it is not as elegant as what you have going here.